src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -751,6 +756,117 @@ public Task SuggestIndexMerge()
 
         }
 
+        [RavenAction("/databases/*/indexes/try", "POST", AuthorizationStatus.ValidUser)]
+        public async Task TestJavaScriptIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "TestJavaScriptIndex");
+                if (input.TryGet("Definition", out BlittableJsonReaderObject index) == false)
+                    ThrowRequiredPropertyNameInRequest("Definition");
+
+                var indexDefinition = JsonDeserializationServer.IndexDefinition(index);
+
+                if (indexDefinition.Maps == null || indexDefinition.Maps.Count == 0)
+                    throw new ArgumentException("Index must have a 'Maps' fields");
+
+                indexDefinition.Type = indexDefinition.DetectStaticIndexType();
+
+                if (indexDefinition.Type.IsJavaScript() == false)
+                    throw new UnauthorizedAccessException("Testing indexes is only allowed for JavaScript indexes.");
+
+                if (indexDefinition.Name.StartsWith(Constants.Documents.Indexing.SideBySideIndexNamePrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new ArgumentException(
+                        $"Index name must not start with '{Constants.Documents.Indexing.SideBySideIndexNamePrefix}'. Provided index name: '{indexDefinition.Name}'");
+                }
+
+                var compiledIndex = new JavaScriptIndex(indexDefinition, Database.Configuration);
+
+                var pageSize = GetIntValueQueryString("pageSize", false) ?? defaultPageSizeForTestingJavaScriptIndex;
```

Comment:
That might not be a good idea, if they are filtering in the index.
Why not just get an array of document ids, instead?

Reply:
This is the best way to go by default since you don't need to define anything and is more likely to run into unexpected document structure than to define a set of documents.
I would much rather write the index and maybe set the "pageSize" which might be a misleading name since it isn't the result's pageSize but the input's.
We can support both but I still think this should be the default.
I'm still waiting for requirements, without any I just implemented what I thought is the best approach, I expect @ml054 to come up with more UI stuff later on.